### ISSUE TITLE
Support CircuitOperation(use_repetition_ids=False)

### ISSUE THREAD

**Issue by 95-martin-orion** [quantumlib/qsim#533]
Highlighted in #531 - apparently `CircuitOperation(use_repetition_ids=False)` does not convert properly to qsim due to not finding the associated measurements.

**Comment by 95-martin-orion** [comment]
Yup, looks like we've been relying on `findall_operations_with_gate_type`, which doesn't bother searching inside of `CircuitOperation`s. Modifying that method is probably the right way to resolve this, although because of how qsimcirq handles the output we won't be able to support `repeat-until` without additional changes.

**Comment by 95-martin-orion** [comment]
One additional issue: `findall_operations_with_gate_type` returns a (moment index, op, gate) tuple. For a sub-operation of a `CircuitOperation`, moment index doesn't precisely locate the operation.

**Comment by maffoo** [comment]
I think some specific logic for finding measurement operations would be generically useful. For example, in `cirq-google` we have a [`find_measurements`](https://github.com/quantumlib/Cirq/blob/master/cirq-google/cirq_google/api/v2/results.py#L47) function to find measurements in a circuit which does something similar to what we need here (that function also needs to be updated to handle `CircuitOperation`).
